lib/strings.nix

```diff
@@ -541,9 +541,13 @@ rec {
        escapeNixIdentifier "0abc"
        => "\"0abc\""
   */
-  escapeNixIdentifier = s:
+  escapeNixIdentifier =
+    let
+      matchIdentifier = match "[a-zA-Z_][a-zA-Z0-9_'-]*";
+    in
+    s:
     # Regex from https://github.com/NixOS/nix/blob/d048577909e383439c2549e849c5c2f2016c997e/src/libexpr/lexer.l#L91
-    if match "[a-zA-Z_][a-zA-Z0-9_'-]*" s != null
+    if matchIdentifier s != null
```

Comment:
Non-partial applications of primops are optimized, and compiled regexes are cached. Did you measure this change?

Reply:
Oh I was kind of wondering the same in https://github.com/NixOS/nixpkgs/pull/292945 (which I just merged). Was assuming that @adisbladis checked it to actually be effective, so I'd like confirmation of this too.